lib/azure/armrest/insights/alert_service.rb

```diff
@@ -0,0 +1,23 @@
+# Azure namespace
+module Azure
+  # Armrest namespace
+  module Armrest
+    # Insights namesspace
+    module Insights
+      # Base class for managing alert rules.
+      class AlertService < ResourceGroupBasedService
+        # The provider used in requests when gathering Alert information.
+        attr_reader :provider
```

Comment:
We may want to move this to the base service since it appears in multiple classes. Can be done in another PR